src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOp.java

```diff
@@ -862,13 +862,22 @@ public Block.Builder lower(Block.Builder b, OpTransformer opT) {
                 Block.Builder curr = blocks.get(i);
                 if (isLabelBody) {
                     Block.Builder expression = blocks.get(i + 1);
-                    boolean isDefaultLabel = i == blocks.size() - 2;
-                    Block.Builder nextLabel = isDefaultLabel ? null : blocks.get(i + 2);
+                    boolean isLastLabel = i == blocks.size() - 2;
```

Comment:
This is a bit tricky. The details on when a default case is added with a `throw` are spread between `TransPatterns` and `Lower`. Basically:
* `TransPatterns` will turn an unconditional pattern (see below) into a default label
* `Lower` will add a `default` throwing label, only if one isn't there
* the type of the exception is `IncompatibleClassChangeError` if selector is an enum, or `MatchException` otherwise.

An unconditional pattern is a pattern whose type is guaranteed to always match the selector expression (e.g. because the type of the pattern is a supertype of the type of the selector expression). This property should be captured by `tree.hasUnconditionalPattern`.  In fact, that flag seems to be set also when the switch has a `default` label. So just checking that flag should suffice (but please double check).

Reply:
Actually... this code is about a lowering step... I think the best way would be to automatically insert a `default` label when generating the code model in `ReflectMethods`. At that point, as explained, the compiler knows which switches are "unconditional" and which are not. We could maybe reflect this piece of info in the model, but I think it's actually better to remove edge cases and just add the missing case.

Reply:
Yes, that seems a better approach given the knowledge the compiler has. For switch ops created manually we anyway cannot guarantee Java program behavior, same for other operations too.

FWIW modeling of `if/else if/else` statements will always add an `else` if not explicitly present with the default behavior that yields nothing.

Reply:
This will make testing easier, we simply need to test that the generated code model has default label that throws the right exception.

Reply:
According to java spec, MatchException will be thrown when no label applies and the selector expression is not null, IncompatibleClassChangeError is not mentioned at all. @mcimadamore